src/index.ts

```diff
@@ -264,6 +266,52 @@ export namespace OneSignal {
       }
     }
 
+    /**
+     * Add a callback that fires when the OneSignal user state changes.
+     * Important: When using the observer to retrieve the onesignalId, check the externalId as well to confirm the values are associated with the expected user.
+     */
+    export function addEventListener(
+      event: 'change',
+      listener: (event: UserChangedState) => void,
+    ) {
+      if (!isNativeModuleLoaded(RNOneSignal)) return;
+
+      isValidCallback(listener);
+      RNOneSignal.addUserStateObserver();
+      eventManager.addEventListener<UserChangedState>(
+        USER_STATE_CHANGED,
+        listener,
+      );
+    }
+
+    /** Clears current user state observers. */
+    export function removeEventListener(
+      event: 'change',
+      listener: (event: UserChangedState) => void,
+    ) {
+      if (!isNativeModuleLoaded(RNOneSignal)) return;
+
+      eventManager.removeEventListener(USER_STATE_CHANGED, listener);
+    }
+
+    /** Get the OneSignal Id associated with the user. */
```

Comment:
Let's point out the nullability of these two IDs within the doc comment for easy access to developers.